components/pointsbox.tsx

```diff
@@ -1,37 +1,42 @@
-import React from 'react';
-import Select from 'react-select';
-import { Option } from './textbox'
+import React from "react";
+import Select, { SingleValue } from "react-select";
+import { AssessmentType } from "../types/Types";
+
+interface Option {
+  value: number;
+  label: string;
+}
 
 interface Pointsboxprop {
-    maxPoints: number;
-    selectedOption: Option | null | undefined;
-    setSelectedOption: any;
+  assessment: AssessmentType;
+  setAssessment: (assessment: AssessmentType, newScore: number) => void;
 }
 
 const Pointsbox: React.FC<Pointsboxprop> = (props: Pointsboxprop) => {
-  const options: Option[] = [{value: 0, label: '0 p'}];
+  const options: Option[] = [{ value: 0, label: "0 p" }];
 
   // add option objects to option list
-  for (let i=1 ; i<props.maxPoints + 1 ; i++) {
-    options.push({value: i, label: i.toString() + ' p'})
+  for (let i = 1; i < props.assessment.maxPoints + 1; i++) {
+    options.push({ value: i, label: i.toString() + " p" });
   }
 
-
+  const handleChange = (selectedOption: Option) => {
+    props.setAssessment(props.assessment, selectedOption.value);
```

Comment:
⚠️  Creates problem when a user removes the selected score.
> Cannot read properties of null (reading 'value')

I think this is better in the MUI-component that I've been working on, so maybe we'll leave this bug.. I think we agreed on that earlier, we just need to remember it. Maybe create an issue in it?